lib/async/http/faraday/adapter.rb

```diff
@@ -168,22 +173,29 @@ def call(env)
 				
 				def perform_request(env)
 					with_client(env) do |endpoint, client|
+						if headers = env.request_headers
+							headers = ::Protocol::HTTP::Headers[headers]
+							
+							# Use content-length to inform body length if given, but remove the header since it will be
+							# set for us later anyway, and not doing so could result in a duplicate content-length headers
+							# if capitalization differs
+							content_length = headers.delete("content-length")&.to_i
+						end
+						
 						if body = env.body
 							# We need to ensure the body is wrapped in a Readable object so that it can be read in chunks:
 							# Faraday's body only responds to `#read`.
 							if body.is_a?(::Protocol::HTTP::Body::Readable)
 								# Good to go
```

Comment:
Should this case try to rebuild the body with the given length too?

Reply:
No I wouldn’t worry about this case either, it’s expected the body would know its size already if appropriate.

The main issue we are addressing is Faradays opaque “body#read” which may have length but the only way to know that is via the supplied header.